chromium_src/third_party/blink/renderer/core/loader/frame_fetch_context.cc

```diff
@@ -0,0 +1,24 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "../../../../../../../third_party/blink/renderer/core/loader/frame_fetch_context.cc"
+
+namespace blink {
+
+String FrameFetchContext::GetCacheIdentifierIfCrossSiteSubframe() const {
+  if (GetResourceFetcherProperties().IsDetached())
+    return cache_identifier_if_cross_site_subframe_;
+
+  String cache_identifier;
+  if (document_->domWindow()->IsCrossSiteSubframeIncludingScheme()) {
```

Comment:
> `IsCrossSiteSubframeIncludingScheme()` uses HTML's definition of "site" as a registrable domain and scheme.
https://github.com/chromium/chromium/blob/98.0.4695.1/third_party/blink/renderer/core/frame/local_dom_window.h#L415-L416

I this this is very similar to what we do in `CookieSettingsBase::ShouldUseEphemeralStorage`, thus we can use this method as is. But if I'm wrong please tell me.
cc @pes10k